core/sdk-core/src/main/java/software/amazon/awssdk/core/client/config/ClientOverrideConfiguration.java

```diff
@@ -392,6 +474,11 @@ default Builder retryPolicy(RetryMode retryMode) {
          * <b>The SDK will not automatically close the executor when the client is closed. It is the responsibility of the
          * user to manually close the executor once all clients utilizing it have been closed.</b>
          *
+         * <p>
+         * When modifying this option from an {@link SdkPlugin}, it is strongly recommended to decorate the
```

Comment:
E.g. this plugin would result in a resource leak:

```java
config -> {
    config.scheduledExecutorService(Executors.newScheduledExecutor(50));
}
```

The reason is because it overrides `config.scheduledExecutorService()`, without shutting it down.

Reply:
Ah I see, that makes sense. I'm guessing there's no good way for us to catch such mistakes internally and close the overridden value?